ray-operator/controllers/ray/common/service.go

```
@@ -376,14 +371,26 @@ func setLabelsforUserProvidedService(service *corev1.Service, labels map[string]
 // getServicePorts will either user passing ports or default ports to create service.
 func getServicePorts(cluster rayv1.RayCluster) map[string]int32 {
 	ports := getPortsFromCluster(cluster)
-	// Assign default ports
-	if len(ports) == 0 {
-		ports = getDefaultPorts()
+
+	defaultPorts := getDefaultPorts()
+	usedPorts := make(map[int32]bool) // Track assigned port values
+
+	for _, port := range ports {
+		usedPorts[port] = true
 	}
 
-	// check if metrics port is defined. If not, add default value for it.
-	if _, metricsDefined := ports[utils.MetricsPortName]; !metricsDefined {
-		ports[utils.MetricsPortName] = utils.DefaultMetricsPort
+	// 1. If a value is provided in the YAML file, overwrite the default value
+	// 2. If no port value is provided, use the default name and port
+	// 3. If a port value is provided in the YAML file, retain the original name
+
+	for name, defaultPort := range defaultPorts {
+		if _, defined := ports[name]; !defined {
+			// Only assign defaultPort if the port is not already in use
+			if !usedPorts[defaultPort] {
+				ports[name] = defaultPort
+				usedPorts[defaultPort] = true // Mark this port as used
+			}
```

Should return error instead of do nothing if ports conflict with defaultPort.

---

What I was think here is that if user set the port with different name. For instance, the default is:

```yaml
- containerPort: 6379
  name: gcs-server
```

While user set:

```yaml
- containerPort: 6379
  name: gcs
```

In this case, user's setting with name gcs will be applied.

Do we need error for this? Or probably a warning is enough?

---

Update: 
This problem is actually caused by different naming for `gcs-server` (some named it as `gcs`)

For example:
https://github.com/ray-project/kuberay/blob/e67237662bb20b9cd3fabf649679e6368be8293b/ray-operator/controllers/ray/common/service_test.go#L71-L72


Default gcs-server name should be:
https://github.com/ray-project/kuberay/blob/e67237662bb20b9cd3fabf649679e6368be8293b/ray-operator/controllers/ray/utils/constant.go#L74

I would like to confirm if it is ok to change all port name "gcs" into "gcs-server"?

---

Yes. We should use the same value across the codebase.